control-plane/catalog/to-consul/resource_test.go

```diff
@@ -1266,6 +1266,39 @@ func TestServiceResource_clusterIPTargetPortNamed(t *testing.T) {
 	})
 }
 
+// Test target information are set in service meta.
+func TestServiceResource_targetRefInMeta(t *testing.T) {
+	t.Parallel()
+	client := fake.NewSimpleClientset()
+	syncer := newTestSyncer()
+	serviceResource := defaultServiceResource(client, syncer)
+	serviceResource.ClusterIPSync = true
+
+	// Start the controller
+	closer := controller.TestControllerRun(&serviceResource)
+	defer closer()
+
+	// Insert the service
+	svc := clusterIPService("foo", metav1.NamespaceDefault)
+	_, err := client.CoreV1().Services(metav1.NamespaceDefault).Create(context.Background(), svc, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// Insert the endpoints
+	createEndpoints(t, client, "foo", metav1.NamespaceDefault)
+
+	// Verify what we got
+	retry.Run(t, func(r *retry.R) {
+		syncer.Lock()
+		defer syncer.Unlock()
+		actual := syncer.Registrations
+		require.Len(r, actual, 2)
+		require.Equal(r, "foobar", actual[0].Service.Meta[ConsulK8SRefValue])
+		require.Equal(r, "pod", actual[0].Service.Meta[ConsulK8SRefKind])
+		require.NotContains(r, ConsulK8SRefValue, actual[0].Service.Meta)
+		require.NotContains(r, ConsulK8SRefKind, actual[0].Service.Meta)
```

Comment:
What is this testing?

Reply:
Actually it was a mistake, I redid it and added a comment. It was supposed to test that the second service didn't contain incorrect refs